Module 6

aaronchan32 opened the pull request:
## Tracking Info

Resolves #45

## Changes

- Added all sections in module
- Added animations for Crash Protocol section
- Updated ModuleButtons component so that it is horizontally fixed

## Testing

http://localhost:3000/module6/

## Confirmation of Change

![image](https://github.com/user-attachments/assets/deb8f7c3-673a-419e-a48d-be538dcf358b)

km322 commented:
Hey Aaron amazing job!

A couple of comments I have:
- Can you wrap the sidebar in a div like we did for module 1  so it stays the same size for every module
```
<div className={styles.hideModile}>
        <Sidebar />
</div>
```
and...
```
@media (max-width: 768px) {
  .sidebar {
    display: none;
  }
}
```
- Another thing is that the bottom buttons have a little bit of different behaviors when zooming in and out. It looks like it goes to the bottom of the page but I thought it should stay underneath the content like in the previous page. This is probably a small thing you can ignore if you disagree.  

https://github.com/user-attachments/assets/28be2076-f537-48f4-81a2-af4859a5b522

Also Quick Questions:
- are the progress bar and next and back buttons supposed to get smaller as you zoom out?
- One thing I notices is that when I spam click the next button, the animation skips. Is this fine?

https://github.com/user-attachments/assets/009092b0-89fd-4ce8-9b9a-98ae8c4ee5ad

Amazing job!

km322 commented:
Hey Aaron! Perfect, everything looks amazing!

the main point of the div is to fix this issue where the sidebar is cut off as shown below. I think wrapping it in a div will fix this but if you have an alternative solution that would be amazing!

https://github.com/user-attachments/assets/1c82fe3a-7a9b-4cc8-ad3d-71747e22ed5d

Thanks!

OrigamiStarz reviewed (commented):
Hey Aaron, nice work, especially with the animation! Everything looks good to me.

I did notice a few minor things:
1. The progress bar at the top does not always fill in correctly. With my Google Chrome profile, it seems like the more I zoom out, the less aligned the "filled in" part is (see video below). I think this is somewhat related to how I have a default large font size in my Chrome settings, as I tested this out on a normal profile and found that it required more zoom to see a difference. However, I feel like most users will not encounter this issue (and even if they do, it does not affect the content), so it might not be worth fixing in the first place. 

https://github.com/user-attachments/assets/793c716e-73a0-4dd8-b2df-d054095fb980

2. Something that struck out to me while reviewing module 6 is how the hyperlink is black, as opposed to the light blue used in module 7. I was thinking it might be worth asking the designers to confirm that they are supposed to be different, as opposed to a universal hyperlink color. 

<img width="738" alt="Screenshot 2025-05-20 at 6 03 38 PM" src="https://github.com/user-attachments/assets/2fdaeb51-6683-4b5d-a54e-3eb38bebef15" />

<img width="869" alt="Screenshot 2025-05-20 at 6 04 17 PM" src="https://github.com/user-attachments/assets/60dc32da-61e1-4d62-bbf4-208f53a34292" />

I'll approve this after you fix the sidebar as Ketan mentioned. Great work!

aaronchan32 commented:
Thanks for the feedback! I fixed the `Sidebar` component and also fixed the bug with the progress bar. I also decided to completely redo the implementation for the progress bar so there should no longer be a flicker after the initial re-render.

I reached out to Justin about the hyperlink inconsistency and will get back to you when he responds.

km322 commented on frontend/src/app/module6/page.tsx:
```diff
@@ -0,0 +1,51 @@
+import BicycleSafetyFacts from "./components/BicycleSafetyFacts/BicycleSafetyFacts";
+import CrashProtocol from "./components/CrashProtocol/CrashProtocol";
+
+export default function Module2() {
```
can you rename the module to module6

aaronchan32 commented:
> Hey Aaron, when testing the code out in Safari I face loads of issues when using svgs, is it possible to make it work in safari or other browsers better?
> 
>  Screen.Recording.2025-05-21.at.1.23.22.PM.mov 
> This also occurs in the home page for safari:
> 
>  Screen.Recording.2025-05-21.at.1.25.59.PM.mov

Hey Ketan, thanks for bringing this up. I unfortunately use a Windows laptop, so it's a bit difficult for me to debug on Safari at the moment. However, once I get my GitHub student re-verification approved I will have access to Safari emulation software that will make this process easier. I will update you once I get this all fixed!

eshaan-s18 reviewed (dismissed):
Fantastic job Aaron! This looks amazing! I approve this PR. Also there are a couple of merge conflicts, if you could resolve those that would be awesome and super helpful! Thanks for all your hard work on this!!

km322 reviewed (approved):
Great job!

OrigamiStarz reviewed (dismissed):
Hey Aaron, everything looks good to me! Just one very minor thing I noticed: 
The text isn't vertically centered well, so when I use a font size of large/very large through chrome settings, it looks off-centered (see below). 
<img width="1440" alt="Screenshot 2025-05-30 at 3 39 59 PM" src="https://github.com/user-attachments/assets/bc7c55d8-4b5b-481a-a914-338c374ad7c9" />

However, I think this won't affect most people, so I'll approve it. Great work!
